Find404.scala

```diff
@@ -1,16 +1,20 @@
 import java.io._
 import scala.io.Source._
 import scala.io.StdIn._
+import scala.collection.mutable.ArrayBuffer
 
-object Find404 {
-    
-    def main(args: Array[String]): Unit = {
-        println("Provide a directory")
-        val directory = new File(readLine())
+val fileSet = ArrayBuffer[File]()
 
-        val directorySet = directory.listFiles()
-        val fileSet = directorySet.filter(_.getClass().getName().indexOfSlice(".") == -1)
+try {
+    val directorySet = new File(args(0))
+    exploreDirect(directorySet, fileSet)
+} catch {
+    case _: NullPointerException => println("Not a valid directory")
+}
 
-        fileSet.foreach(println(_))
+def exploreDirect(directory: File, fileSet: ArrayBuffer[File]): Unit = {
```

Comment:
Instead of returning a unit, return an array of files to the caller so it can be the responsibility of the caller to decide what to do with all the files.

By making it return unit you're saying that this function ultimately will do all the work and your application logic will be constraint to be running inside of this function.

If you return an array of files that match some criteria, another person can iterate on that and filter more, or map it, or do something foreach. Your program becomes easier to reason about and your code is more generally useful (reusable).